eng/helix/Helix.proj

```diff
@@ -0,0 +1,164 @@
+    <QueueSuffix>.open</QueueSuffix>
+    <!-- Open queues require a creator to be set. -->
+    <Creator Condition=" '$(USERNAME)' != '' ">$(USERNAME)</Creator>
+    <Creator Condition=" '$(USER)' != '' ">$(USER)</Creator>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(SYSTEM_COLLECTIONURI)' == 'https://dev.azure.com/dnceng/'">
+    <HelixSourcePrefix Condition="'$(BUILD_REASON)' == 'Manual'">pr</HelixSourcePrefix>
+    <HelixSourcePrefix Condition="'$(BUILD_REASON)' == 'PullRequest'">pr</HelixSourcePrefix>
+    <HelixSourcePrefix Condition="'$(HelixSourcePrefix)' == ''">official</HelixSourcePrefix>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <HelixSourcePrefix Condition="'$(HelixSourcePrefix)' == ''">pr</HelixSourcePrefix>
+    <HelixSource>$(HelixSourcePrefix)/dotnet/dotnet-monitor</HelixSource>
+    <HelixSource Condition="'$(BUILD_SOURCEBRANCH)' != ''">$(HelixSource)/$(BUILD_SOURCEBRANCH)</HelixSource>
+    <HelixType>test/binaries/</HelixType>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <HelixAvailableTargetQueue>
+      <AdditionalProperties>Platform=$(HelixArchitecture)</AdditionalProperties>
+    </HelixAvailableTargetQueue>
+  </ItemDefinitionGroup>
+
+  <!-- List of all available Helix machines: https://helix.dot.net/ -->
+
+  <!-- Windows queues -->
+  <ItemGroup Condition="$([MSBuild]::IsOSPlatform(Windows))">
+    <!-- win-arm64 -->
+    <HelixAvailableTargetQueue Include="windows.11.arm64$(QueueSuffix)"
+                               Condition="'$(HelixArchitecture)' == 'arm64'">
+      <TestRunName>Windows 11 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+    </HelixAvailableTargetQueue>
+    <!-- win-x64 -->
+    <HelixAvailableTargetQueue Include="windows.10.amd64.client$(QueueSuffix)"
+                               Condition="'$(HelixArchitecture)' == 'x64'">
+      <TestRunName>Windows 10 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+    </HelixAvailableTargetQueue>
+    <!-- win-x86 -->
+    <!-- Windows x64 clients can run dotnet x86 natively -->
+    <HelixAvailableTargetQueue Include="windows.10.amd64.client$(QueueSuffix)"
+                               Condition="'$(HelixArchitecture)' == 'x86'">
+      <TestRunName>Windows 10 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+    </HelixAvailableTargetQueue>
+  </ItemGroup>
+
+  <!-- Linux gnu libc queues -->
+  <ItemGroup Condition="$([MSBuild]::IsOSPlatform(Linux)) and !$(PackageRid.Contains(musl))">
+    <!-- linux-arm64 -->
+    <HelixAvailableTargetQueue Include="ubuntu.1804.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-arm64v8"
+                               Condition="'$(HelixArchitecture)' == 'arm64'">
+      <TestRunName>Debian 11 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+    </HelixAvailableTargetQueue>
+    <HelixAvailableTargetQueue Include="ubuntu.1804.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8"
```

Comment:
I know you mentioned this the other day that 18.04 was realistically the oldest version we should be running against - would there be any benefit to also running against a newer version (and I think the same question goes for macOS, but for an older version)?

Reply:
I think that's a reasonable suggestion (use the oldest supported and newest support version of a distro). I'm going to hold off on this for now and get the tests running first so that way we can see how fast these things run in practice; we might even want to limit the set for PRs, do more for CI, and all variations on a scheduled basis.

Reply:
#4911